Use relative DNS for clusters with non-default clusterDomain

harmjanblok opened the pull request:
Kubernetes clusters can override the clusterDomain, which defaults to `cluster.local`. For clusters using a non-default clusterDomain the value of `ZEEBE_BROKER_NETWORK_ADVERTISEDHOST` and `ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS` are incorrect.

A default `resolv.conf` for a Pod contains following `search`:

```bash
search $NAMESPACE.svc.$CLUSTER_DOMAIN svc.$CLUSTER_DOMAIN $CLUSTER_DOMAIN
```

Hence the `$CLUSTER_DOMAIN` part can be omitted from the mentioned environment variables.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda-cloud-helm?pullRequest=234) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda-cloud-helm?pullRequest=234) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda-cloud-helm?pullRequest=234) it.</sub>

Zelldon reviewed (commented):
Thanks @harmjanblok this is an interesting to learn ! I have to check your changes whether it still works. 

Please make sure to follow our commit guidelines https://github.com/camunda/camunda-cloud-helm/blob/main/CONTRIBUTING.md#commit-guidelines

I will come back after I verified that :) In the meantime could you please run `make golden` to generate the new golden files ? 

Greets
Chris

Zelldon commented:
Hey @harmjanblok our team installed a new CLA bot version and it seems that all previous CLA signatures are lost, do you mind to sign the CLA again?

harmjanblok commented:
Hi @Zelldon, I've signed the CLA (again), updated the commit message as suggested (I would consider it a feature?) and rebased on main. Still working on the `make golden`...

harmjanblok commented:
I needed to run `make deps` before running `make golden`.

Zelldon commented:
Thanks @harmjanblok :+1: 

> I needed to run make deps before running make golden.

Good point! I will adjust the makefile then

Zelldon reviewed (approved):
Sorry for the delay! I tested it today as far as I can say it works fine :) thanks for the changes :bow:

Zelldon commented:
Part of https://github.com/camunda/camunda-platform-helm/releases/tag/ccsm-helm-0.0.26
